examples/remote/basic.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2023 MetPy Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+==================
+Remote Data Access
+==================
+
+Use MetPy to access data hosted in known AWS S3 buckets
+"""
+from datetime import datetime, timedelta
+
+from metpy.remote import GOESArchive, NEXRADLevel2Archive, NEXRADLevel3Archive
+
+###################
+# NEXRAD Level 2
+
+# Get the nearest product to a time
+prod = NEXRADLevel2Archive().get_product('KTLX', datetime(2013, 5, 22, 21, 53))
+
+# Open using MetPy's Level2File class
+l2 = prod.parse()
+
+###################
+# NEXRAD Level 3
+start = datetime(2022, 10, 30, 15)
+end = start + timedelta(hours=2)
+products = NEXRADLevel3Archive().get_range('FTG', 'N0B', start, end)
+
+# Get all the file names--could also get a file-like object or open with MetPy Level3File
+print([prod.name for prod in products])
+
+################
+# GOES Archives
+prod = GOESArchive(16).get_product('ABI-L1b-RadC', band=2)
```

Comment:
With GOES16 no longer in production, this fails with an ambiguous,
```pytb
ValueError: min() iterable argument is empty
```
out of `_get_closest`. `_build_time_prefix` seems to only search same day. Does this also risk failure when searching shortly after 0000 UTC, depending on when data goes live?

Reply:
Actually, I think it's only searching the same hour. I've had thoughts that maybe we could extend the approach taken for NEXRAD level 3, where we iteratively dig down, bounding our search for the closest, across all of the archives. I would suggest leaving that as an exercise for the future, though.

I have caught that error and raised an improved error message for that case now.

Reply:
Also updated the example to use GOES 19.